src/RevMem.cc

```diff
@@ -372,8 +372,9 @@ bool RevMem::ReadMem(uint64_t Addr, size_t Len, void *Target,
 
       // Read data
       BaseMem = &physMem[tmpPhysAddr];
-      ctrl->sendREADRequest(Addr, (uint64_t)(BaseMem), tmpLen, Target, flags);
-
+      ctrl->sendREADRequest(Addr, (uint64_t)(BaseMem), tmpLen, DataMem, flags);
+      DataMem = DataMem + lenCount;
```

Comment:
why are you incrementing DataMem

Reply:
> why are you incrementing DataMem

The memory read/write operation is implemented on a page basis. Increasing DataMem is to move to the target on the next page.